gdk/PixbufLoader.custom

```diff
@@ -141,3 +147,30 @@
 		{
 			return new PixbufLoader (System.Reflection.Assembly.GetCallingAssembly (), resource);
 		}
+
+        class ByteArrayPool
+        {
+            WeakReference<System.Collections.Concurrent.ConcurrentQueue<byte[]>> queuedBuffers =
+                new WeakReference<System.Collections.Concurrent.ConcurrentQueue<byte []>> (new System.Collections.Concurrent.ConcurrentQueue<byte []> ());
+
+            public byte[] Take()
+            {
+                System.Collections.Concurrent.ConcurrentQueue<byte[]> queue;
+                byte [] result;
+
+                if (queuedBuffers.TryGetTarget (out queue) && queue.TryDequeue (out result))
+                    return result;
+
+                return new byte [bufSize];
+            }
+
+            public void Put(byte[] array)
+            {
+                System.Collections.Concurrent.ConcurrentQueue<byte []> queue;
+                if (!queuedBuffers.TryGetTarget (out queue)) {
+                    queue = new System.Collections.Concurrent.ConcurrentQueue<byte []> ();
+                    queuedBuffers.SetTarget (queue);
```

Comment:
Do we need to protect against multi-threaded access here? and maybe in Take() as well?